File: android/src/astro/axis/planet/libgdx/helper/SwipeGestureListener.java
package astro.axis.planet.libgdx.helper;

import android.view.GestureDetector;
import android.view.MotionEvent;
import android.app.Activity;


public class SwipeGestureListener extends GestureDetector.SimpleOnGestureListener {

    private static final int SWIPE_MIN_DISTANCE = 200;
    private static final int SWIPE_THRESHOLD_VELOCITY = 200;

    private final Activity activity;

    public SwipeGestureListener(Activity activity) {
        this.activity = activity;
    }

    @Override
    public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {

        if (e2.getX() - e1.getX() > SWIPE_MIN_DISTANCE && Math.abs(velocityX) > SWIPE_THRESHOLD_VELOCITY) {
            activity.finish();
        }

        return false;
    }
}

File: android/src/astro/axis/planet/libgdx/helper/SwipeGestureListener.kt
package astro.axis.planet.libgdx.helper

import android.app.Activity
import android.view.GestureDetector.SimpleOnGestureListener
import android.view.MotionEvent
import kotlin.math.abs

class SwipeGestureListener(private val activity: Activity) : SimpleOnGestureListener() {
    override fun onFling(e1: MotionEvent?, e2: MotionEvent, velocityX: Float, velocityY: Float): Boolean {
        if (e2.x - e1!!.x > SWIPE_MIN_DISTANCE && abs(velocityX.toDouble()) > SWIPE_THRESHOLD_VELOCITY) {
            activity.finish()
        }

        return false
    }

    companion object {
        private const val SWIPE_MIN_DISTANCE = 200
        private const val SWIPE_THRESHOLD_VELOCITY = 200
    }
}
